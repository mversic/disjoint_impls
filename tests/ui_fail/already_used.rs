use disjoint_impls::disjoint_impls;

pub trait Dispatch {
    type Group;
}

disjoint_impls! {
    pub trait Kita {
        type Kara<T>;
        fn kita<T>(a: T);
    }

    impl<T: Dispatch<Group = impl Dispatch>> Kita for T {
        type Kara<T> = ();
        fn kita<T>(a: T) {}
    }
}

fn main() {}
