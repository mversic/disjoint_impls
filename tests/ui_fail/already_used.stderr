error: `impl Trait` is not allowed in this position
  --> tests/ui_fail/already_used.rs:13:30
   |
13 |     impl<T: Dispatch<Group = impl Dispatch>> Kita for T {
   |                              ^^^^^^^^^^^^^
